profitbricks.go

```diff
@@ -1,50 +1,102 @@
 package profitbricks
 
-import "strconv"
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"gopkg.in/resty.v1"
+)
+
+var DebugHTTP = false
 
-//Client object
 type Client struct {
-	client *client
+	*resty.Client
+	// AuthApiUrl will be used by methods talking to the auth api by sending absolute urls
+	AuthApiUrl  string
+	CloudApiUrl string
 }
 
-//NewClient is a constructor for Client object
-func NewClient(username, password string) *Client {
-	c := newPBRestClient(username, password, "", "", "", true)
-	return &Client{
-		client: c,
+var Version = "5.0.2"
+
+const (
+	DefaultApiUrl  = "https://api.ionos.com/cloudapi/v5"
+	DefaultAuthUrl = "https://api.ionos.com/auth/v1"
+)
+
+func RestyClient(username, password, token string) *Client {
+	c := &Client{
+		Client:      resty.New(),
+		AuthApiUrl:  DefaultAuthUrl,
+		CloudApiUrl: DefaultApiUrl,
 	}
+	if token == "" {
+		c.SetBasicAuth(username, password)
+	} else {
+		c.SetAuthToken(token)
+	}
+	c.SetDebug(DebugHTTP)
+	c.SetHostURL(DefaultApiUrl)
+	c.SetDepth(5)
+	c.SetUserAgent("ionos-enterprise-sdk-go " + Version)
+	c.SetRetryCount(1)
+	c.AddRetryCondition(
+		func(r *resty.Response) (bool, error) {
+			if r.StatusCode() == http.StatusTooManyRequests {
+				retryAfter := r.Header().Get("Retry-After")
+				var dur time.Duration
+				var err error
+				if retryAfter != "" {
+					dur, err = time.ParseDuration(r.Header().Get("Retry-After") + "s")
+					if err != nil {
+						return false, err
+					}
+				} else {
+					dur = 1 * time.Second
```

Comment:
X-RateLimit-Remaining will grow by one each second until its value equals X-RateLimit-Burst. Therefore I think waiting only one second is not too fast.

Reply:
I don't understand your argument.

If we encounter this error and then retry every second we consume all newly available requests. So we actively try to keep the situation of having run out of requests instead of waiting a moment so that more requests become available.